internal/services/eventhub/eventhub_resource.go

```diff
@@ -114,24 +120,46 @@ func resourceEventHub() *pluginsdk.Resource {
 										Required: true,
 										ValidateFunc: validation.StringInSlice([]string{
 											"EventHubArchive.AzureBlockBlob",
+											"EventHubArchive.AzureDataLake",
 											// TODO: support `EventHubArchive.AzureDataLake` once supported in the Swagger / SDK
 											// https://github.com/Azure/azure-rest-api-specs/issues/2255
 											// BlobContainerName & StorageAccountID can then become Optional
 										}, false),
 									},
+
 									"archive_name_format": {
 										Type:         pluginsdk.TypeString,
 										Required:     true,
 										ValidateFunc: validate.ValidateEventHubArchiveNameFormat,
 									},
+
 									"blob_container_name": {
-										Type:     pluginsdk.TypeString,
-										Required: true,
+										Type:          pluginsdk.TypeString,
+										Optional:      true,
+										AtLeastOneOf:  []string{"capture_description.0.destination.0.blob_container_name", "capture_description.0.destination.0.datalake_account_name"},
+										ConflictsWith: []string{"capture_description.0.destination.0.datalake_account_name"},
 									},
+
 									"storage_account_id": {
 										Type:         pluginsdk.TypeString,
-										Required:     true,
+										Optional:     true,
 										ValidateFunc: azure.ValidateResourceID,
+										RequiredWith: []string{"capture_description.0.destination.0.blob_container_name"},
+									},
+
+									// can not use the id since service won't return resource group data
```

Comment:
how does the eventhub service know what resource group the data lake is in?

Reply:
it doesn't have to know the RG, just as it doesn't capture RG information for storage account:
      blob_container_name = azurerm_storage_container.test.name

The capture api locates the resource by its unique name.

Reply:
can you not have two datalakes with the same name in two different resource groups?

Reply:
Based on my understanding of the service, datalake, servicebus and eventhub namespace should all have the same naming standard. The backend service is locating the resource based on the unique endpoint like below examples:

for service bus namespace
serviceBusEndpoint": "https://acctest2-220616093346299338.servicebus.windows.net:443/",
for eventhub namespace:
"serviceBusEndpoint": "https://xiaxintest-ehn-standard.servicebus.windows.net:443/",
for datalake store:
![image](https://user-images.githubusercontent.com/92154856/184810031-9ff0b14f-eee5-4850-ae2c-8c383de2aa21.png)

So the answer is negative, you can't have two datalake store with exactly the same name. Hope it makes sense to you.